crates/swc_core/Cargo.toml

```diff
@@ -0,0 +1,87 @@
+[package]
+authors = ["강동윤 <kdy1997.dev@gmail.com>", "OJ Kwon <kwon.ohjoong@gmail.com>"]
+description = "TBD"
+name = "swc_core"
+license = "Apache-2.0"
+edition = "2021"
+repository = "https://github.com/swc-project/swc.git"
+version = "0.0.1"
+
+[features]
+### Public features
+### Users should opt in necessary features explicitly
+
+## General
+# Enable `quote!` macro support.
+quote = ["swc_ecma_quote"]
+
+## Plugins
+# Top level features should be enabled to write plugins for the custom transform.
+plugin_transform = [
+  # Dependent features
+  "__plugin_transform",
+  "__common",
+  "__ast",
+  "__visit",
+  "__plugin_transform_schema_v1",
+  # Enable optional packages
+  "swc_common/plugin-mode",
+  "swc_plugin_proxy/plugin-mode",
+  "swc_plugin_macro",
+  "swc_plugin",
+  "once_cell",
+]
+
+### Internal features that public features are relying on.
+### This is not supposed to be used directly, and does not gaurantee
+### stability across each versions.
+
+# Specify version of AST schema will be used.
+# This'll be automatically selected via transform_plugin features,
+# SWC upstream decides which version to be used for specific version of
+# swc_core.
+__plugin_transform_schema_v1 = ["swc_common/plugin_transform_schema_v1"]
+
+# Do not use: testing purpose only
+__plugin_transform_schema_vtest = ["swc_common/plugin_transform_schema_vtest"]
+
+## Plugins
+
+# Internal flags for any transform plugin feature
+__plugin_transform = []
+
+# Do not use: testing purpose only
+# Force enable different version of AST schema
+__plugin_transform_schema_test = [
+  # Dependent features
+  "__plugin_transform",
+  "__common",
+  "__ast",
+  "__visit",
+  "__plugin_transform_schema_vtest",
+  # Enable optional packages
+  "swc_common/plugin-mode",
+  "swc_plugin_proxy/plugin-mode",
+  "swc_plugin_macro",
+  "swc_plugin",
+  "once_cell",
+]
+
+## Common
+__common = ["swc_common"]
```

Comment:
I think this should depend on `swc_atoms` too.
(`swc_common` depends on `swc_atoms`

Reply:
Nevermind. User should depend on `__ast` anyway

Reply:
We don't need to list all the transitive deps to the top level for the specific feature, isn't it? We can easily add later once we find those dependencies around reexports.